nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -57,17 +57,13 @@ let
         };
 
         log-driver = mkOption {
-          type = types.str;
-          default = "none";
+          type = types.enum ["none" "json-file" "syslog" "journald" "gelf" "fluentd" "awslogs" "splunk" "etwlogs" "gcplogs"];
+          default = config.virtualisation.docker.logDriver;
```

Comment:
Oh, I like this!  Deriving the default from whatever the docker daemon is configured to do is probably better than having a hardcoded default for declarative containers in here.  To make this simpler, perhaps make this option a `nullOr str`, with `null` being the default, and only include a `--log-driver=` argument if this option has a non-default value?

Reply:
Done.